src/commands/ext-dev-publish.ts

```diff
@@ -21,7 +25,16 @@ marked.setOptions({
  */
 export const command = new Command("ext:dev:publish <extensionRef>")
   .description(`publish a new version of an extension`)
-  .option(`-s, --stage <stage>`, `release stage (supports "rc", "alpha", "beta", and "stable")`)
+  .option(`-s, --stage <stage>`, `release stage (supports "alpha", "beta", "rc", and "stable")`)
+  .option(
+    `--repo <repo>`,
+    `Public Git repo URI (only required for first version, cannot be changed)`
+  )
+  .option(`--ref <ref>`, `commit hash, branch, or tag to build from the repo (defaults to HEAD)`)
```

Comment:
Seems like it doesn't actually default to HEAD atm - if it is omitted, the code here will treat it like a local source.

Reply:
This is working as intended. We will need to wait for a major CLI version to change the default behavior. For now, --ref has to be specified for repo publishing even though there's a prompt for it.

Reply:
I'm not sure this is WAI in all cases. Right now, this says it defaults to HEAD. However, if i run `firebase ext:dev:publish --repo=github.com/my/repo --root=myroot`,
it will attempt to publish the local directory, even though I passed in multiple flags to publish a remote source.

Reply:
Ah I see, you're right, good catch. I misunderstood what you meant at first. Changed it to throw an error if `ref` isn't specified, since we still want to default the behavior to local source until a major version.